pkg/spec/spec.go

```diff
@@ -47,8 +47,10 @@ func Parse(data []byte, name string) (*v1alpha1.Config, error) {
 		return nil, errors.Wrap(err, "parsing spec.json")
 	}
 
-	// set the name field
-	config.Metadata.Name = name
+	// set the name field if not given in spec
+	if config.Metadata.Name == "" {
```

Comment:
Don't do it this way. The filepath is Tanka's source of truth. Instead, just sanitize the filepath, by substituting all `/` with `-`, as my original PR did:
https://github.com/grafana/tanka/blob/2f61573f7d8df6ec2dff6b3dbf3da8a09ce12eb4/pkg/spec/v1alpha1/config.go#L38-L40

The fact we have names in the `spec.json` is just a bad side-effect from json marshalling, because we need the field to be included when marshalling for passing to Jsonnet, we can't `omitempty` or `-` it.

Reply:
ok, makes sense... sort of. I guess it does root the file system at the ksonnet folder somehow, so that helps it be deterministic I suppose. ` tanka.dev/environment: environments.dns.eu-west2.kube-system` is a little bit of a verbose env name, but that's ok.